include/rocksdb/db.h

```diff
@@ -705,33 +717,42 @@ class DB {
   // benefits.
   // Parameters -
   // options - ReadOptions
-  // column_family - ColumnFamilyHandle* that the keys belong to. All the keys
-  //                 passed to the API are restricted to a single column family
   // num_keys - Number of keys to lookup
+  // column_families - Pointer to C style array of ColumnFamilyHandle* that
+  //                   the keys belong to.
   // keys - Pointer to C style array of key Slices with num_keys elements
   // values - Pointer to C style array of PinnableSlices with num_keys elements
+  // timestamps - Pointer to C style of std::string that, if non-null and
+  //              timestamps are enabled, will have the timestamps of the keys
   // statuses - Pointer to C style array of Status with num_keys elements
   // sorted_input - If true, it means the input keys are already sorted by key
   //                order, so the MultiGet() API doesn't have to sort them
   //                again. If false, the keys will be copied and sorted
   //                internally by the API - the input array will not be
   //                modified
-  virtual void MultiGet(const ReadOptions& options,
-                        ColumnFamilyHandle* column_family,
-                        const size_t num_keys, const Slice* keys,
-                        PinnableSlice* values, Status* statuses,
-                        const bool /*sorted_input*/ = false) {
+  virtual void MultiGet(const ReadOptions& options, const size_t num_keys,
+                        ColumnFamilyHandle** column_families, const Slice* keys,
+                        PinnableSlice* values, std::string* timestamps,
+                        Status* statuses, const bool /*sorted_input*/ = false) {
```

Comment:
These overloads are kind of confusing (which is I guess why we're cleaning them up ;) ) and I might be getting lost here but isn't this the most general overload that others should be implemented in terms of (either directly or indirectly)?

Reply:
Oh, I see it falls back to the non-batched `MultiGet` logic! Again, I think we could consider deprecating those, or if not, consider flipping things around and implementing the older APIs in terms of the newer/better ones

Reply:
Not every derived class implements the batched `MultiGet`. In fact, only `DBImpl` implements it, hence we can't deprecate/redirect the non-batched one to the batched one.